src/in_progress.rs

```diff
@@ -1110,3 +1110,132 @@ pub fn is_hardware_waiting_repair(
+    let stmt_query = conn.query(&stmt, &[&storage_type])?;
+
+    let mut storage_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return storage_id
+        let id: i32 = res.get(0);
+        storage_id = id as u32;
+        debug!(
+            "Storage id {} for the storage_type {}",
+            storage_id, storage_type
+        );
+        Ok(Some(storage_id))
+    } else {
+        // does not exist
+        debug!("No storage with type {} in database", storage_type);
+        Ok(None)
+    }
+}
+
+/// Get storage id based on the storage type.
+pub fn get_storage_detail_id(
+    pool: &Pool<ConnectionManager>,
+    storage_id: u32,
+    region_id: u32,
+    host_name: &str,
+) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get storage detail Id
+    let stmt = "SELECT detail_id FROM storage_details WHERE storage_id = $1
+            AND region_id = $2 AND hostname = $3 ";
+    let stmt_query = conn.query(&stmt, &[&storage_id,&region_id, &host_name])?;
+
+    let mut storage_detail_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return storage_id
+        let id: i32 = res.get(0);
+        storage_detail_id = id as u32;
+        debug!(
+            "Storage details id {} for the host_name {} , region {} , storage_id {} ",
+            storage_detail_id, host_name, region_id, storage_id
+        );
+        Ok(Some(storage_detail_id))
+    } else {
+        // does not exist
+        debug!(
+            "No storage detail id with host_name {} , region {} , storage_id {} 
+        in database",
+            host_name, region_id, storage_id,
+        );
+        Ok(None)
+    }
+}
+
+/// Get a list of ticket IDs (JIRA/other ids) that belong to all servers.
+/// that are in pending state  and outstanding tickets
+pub fn get_all_pendout_tickets(
+    pool: &Pool<ConnectionManager>
+) -> BynarResult<Vec<DiskRepairTicket>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get all tickets of myself with device.state=WaitingForReplacement and operation_detail.status = pending or in_progress
+     let stmt = "SELECT tracking_id, device_name, device_path FROM operation_details JOIN operations
+     USING (operation_id) JOIN hardware USING (device_id) WHERE
+     (status=$1 OR status=$2) AND
+     type_id = (SELECT type_id FROM operation_types WHERE op_name= $3) AND
+     hardware.state in ($4, $5) AND tracking_id IS NOT NULL ORDER BY operations.start_time";
+
+
+    let stmt_query = conn.query(
+        &stmt,
+        &[
+            &OperationStatus::InProgress.to_string(),
+            &OperationStatus::Pending.to_string(),
+            &OperationType::WaitingForReplacement.to_string(),
+            &State::WaitingForReplacement.to_string(),
+            &State::Good.to_string()
+        ],
+    )?;
+    let mut tickets: Vec<DiskRepairTicket> = Vec::new();
```

Comment:
This is fine.  Another possible pattern you could use would be this:
```
if stmt_query.is_empty() {
  debug!(..);
  return Ok(vec![]);
}
debug!("{} pending tickets ... ");
let tickets: Vec<DiskRepairTicket> = stmt_query.iter().map(|row| row_to_ticket(row)).collect();
Ok(tickets)
```
food for thought maybe.  No need to use it if you don't want to.